modules/dnn/src/layers/tile_layer.cpp

```diff
@@ -0,0 +1,99 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "layers_common.hpp"
+
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace cv { namespace dnn {
+
+class TileLayerImpl CV_FINAL : public TileLayer
+{
+public:
+    TileLayerImpl(const LayerParams& params)
+    {
+        setParamsFrom(params);
+        if (params.has("repeats"))
+        {
+            DictValue param_repeats = params.get("repeats");
+            int n_repeats = param_repeats.size();
+
+            CV_Assert(n_repeats > 0);
+            repeats.resize(n_repeats);
+            for (int i = 0; i < n_repeats; i++)
+                repeats[i] = param_repeats.get<int>(i);
+        }
+        else
+            CV_Error(Error::StsNotImplemented, "Tile: repeats needs to be treated as parameter but it is missing.");
+    }
+
+    virtual bool supportBackend(int backendId) CV_OVERRIDE
+    {
+        return backendId == DNN_BACKEND_OPENCV;
+    }
+
+    virtual bool getMemoryShapes(const std::vector<MatShape> &inputs,
+                                 const int requiredOutputs,
+                                 std::vector<MatShape> &outputs,
+                                 std::vector<MatShape> &internals) const CV_OVERRIDE
+    {
+        // inputs[0]: input, a tensor of any shape
+        // inputs[1]: repeats, ignored
```

Comment:
If we don't support it, we should expect 1 input.

Reply:
Indeed one input should be expected, but `repeats` is an input item for Tile. I did not find a example on removing a constant input from the inputs of a layer in the ONNX importer though. Maybe I missed something.

So I just leave it as-is but put it in the list of parameters in addition.

Reply:
You  can see the example here:
https://github.com/opencv/opencv/blob/3f371fe2dd909b29fa6cc01b35649e3b58b265f2/modules/dnn/src/onnx/onnx_importer.cpp#L660
I might've missed something obvious in the API(pop_back or something), but my approach is clear inputs and add new ones manually. In your case that would be clear + add first input.

Reply:
Surprisingly, the number of input (printed in `getMemoryShapes`) is always one regardless of the 'clear + add first input' change.

I also had a try with the tile onnx model with `repeats` being `category: Constant` instead of `category: Initializer`, and the number of input is still one.

Any ideas?

Reply:
Yeah, you're right. Since mentioned inputs don't become constant layers, we get only the first one. Then we just need to change the assert from <=3 to exactly 1.